board/miyoo/boot/ipk_install.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+export TERM=linux
+ipk_directory="/mnt/pkgs"
+package_list=()
+help_messages=()
+index=1
+for ipk_file in "$ipk_directory"/*.ipk; do
+    if [[ -f "$ipk_file" ]]; then
+	ipk_name=$(basename -s .ipk "$ipk_file")
+	package_list+=("$index" "$ipk_name" "off")
+	help_messages[$((2 * index))]="$index"
+	help_messages[$((2 * index + 1))]="$ipk_file"
+	((index++))
+    fi
+done
+if [[ ${#package_list[@]} -eq 0 ]]; then
+    echo "No packages to install."
+    exit 1
+fi
+
+
+until selected_packages=$(dialog --backtitle "MiyooCFW 2.0" --title "IPK package manager" --no-shadow --help-button --help-status --checklist "Press Y to select packages to install:" 20 60 15 "${package_list[@]}" 3>&1 1>&2 2>&3); do
+    case $? in
+	(1) btn=Cancel && break;;
+	(2)
+	    set -- $selected_packages
+	    shift
+	    index=$1
+	    package_info=$(opkg info "${help_messages[$((2 * index + 1))]}")
+	    package_name=$(echo "$package_info" | grep "^Package:" | cut -d' ' -f2)
+	    package_description=$(echo "$package_info" | grep "^Description:" | cut -d' ' -f2-)
+	    dialog --backtitle "MiyooCFW 2.0" --no-shadow --msgbox "$package_name: $package_description" 0 0
+	    for ((i = 2; i < ${#package_list[@]}; i += 3)); do
+		package_list[$i]="off"
+	    done
+	    shift
+	    for selected in "$@"; do
+		package_list[$((3 * selected - 1))]="on"
+	    done
+    esac
+done
+
+clear
+if [[ $? -ne 0 || -z "$selected_packages" ]]; then
+    echo "No packages were selected."
+    exit 1
+fi
+
+for package in $selected_packages; do
```

Comment:
before running opkg install add `dialog --ok-label NO --cancel-label YES` with WARNING statement:
"MiyooCFW team is not responsible for any issues arising from the use of third-party applications with current software. Support, bug fixes, or improvements for 3rd-party apps are not provided by our team members unless explicitly stated otherwise (contact Maintainer of the app).\n Do you still wish to CONTINUE with install?"
For these maybe grep Maintainer from opkg info to Help description